File: kraken/src/main/java/glitch/kraken/object/json/list/Subscribers.java
package glitch.kraken.object.json.list;

import com.google.common.collect.ImmutableList;
import glitch.api.objects.json.interfaces.OrdinalList;
import glitch.kraken.object.json.Subscriber;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;

@Data
public class Subscribers implements OrdinalList<Subscriber> {
    private final ImmutableList<Subscriber> data;
    @Getter(AccessLevel.NONE)
    private final int total;

    @Override
    public int size() {
        return total;
    }
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/collections/Subscribers.kt
package glitch.kraken.`object`.json.collections

import com.google.gson.annotations.SerializedName
import glitch.api.objects.json.interfaces.OrdinalList
import glitch.kraken.`object`.json.Subscriber

data class Subscribers(
        override val data: List<Subscriber>,
        @SerializedName("total")
        override val size: Int = 0
) : OrdinalList<Subscriber>
